src/pybind/mgr/dashboard/frontend/src/app/shared/directives/update-scope.directive.ts

```diff
@@ -0,0 +1,33 @@
+import { Directive, Input, Optional, ElementRef } from '@angular/core';
+import { Permissions } from '../models/permissions';
+import { PermissionScopeDirective } from './permission-scope.directive';
+import { AuthStorageService } from '../services/auth-storage.service';
+
+@Directive({
+  selector: 'input:not([cdNoUpdateScope]), select:not([cdNoUpdateScope])'
+})
+export class UpdateScopeDirective{
+  private permissions: Permissions
+  private service_name: keyof Permissions
+
+  @Input('disabled') disabled: boolean;
+
+  constructor(
+    @Optional() private scope: PermissionScopeDirective, 
```

Comment:
I tried removing the Optional decorator but its giving me NullInjectionError. So I think keeping the Optional() is safer? @epuertat

Reply:
Perhaps this happens because of the order in which these directives are defined? When UpdateScopeDirective is defined, PermissionScopeDirective is not yet defined?

You could either try to alter that order or simply use [a forward-definition mechanism (useful mainly for circular references)](https://blog.thoughtram.io/angular/2015/09/03/forward-references-in-angular-2.html):
```ts
    @Inject(forwardRef(() =>  PermissionScopeDirective)) scope, 
```

Reply:
Yeah, ignore this one too plz. Has nothing to do with the order but with the fact that most input elements don't have any Scope assigned. Optional is perfect here!

Reply:
Perfect.